stdlib/public/Concurrency/Task.cpp

```diff
@@ -534,32 +561,93 @@ static AsyncTaskAndContext swift_task_create_commonImpl(
   }
 
   AsyncTask *parent = nullptr;
+  AsyncTask *currentTask = swift_task_getCurrent();
   if (jobFlags.task_isChildTask()) {
-    parent = swift_task_getCurrent();
+    parent = currentTask;
     assert(parent != nullptr && "creating a child task with no active task");
   }
 
-  // Inherit the priority of the currently-executing task if unspecified and
-  // we want to inherit.
-  if (jobFlags.getPriority() == JobPriority::Unspecified &&
-      (jobFlags.task_isChildTask() || taskCreateFlags.inheritContext())) {
-    AsyncTask *currentTask = parent;
-    if (!currentTask)
-      currentTask = swift_task_getCurrent();
-
-    if (currentTask)
-      jobFlags.setPriority(currentTask->getPriority());
-    else if (taskCreateFlags.inheritContext())
-      jobFlags.setPriority(swift_task_getCurrentThreadPriority());
+  // Start with user specified priority at creation time (if any)
+  JobPriority basePriority = (taskCreateFlags.getCreationPriority());
+
+
+  if (swift_task_is_detached(taskCreateFlags, jobFlags)) {
+    // Case 1: No priority specified
+    //    Base priority = UN
```

Comment:
UN meaning Unspecified, I assume?

Reply:
Yes! Sorry we use this shorthand in Darwin. I'll change the comments here.